lib/mvp2/menu/menu_data/menu_service.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:path/path.dart' as path;
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+import 'package:homebased_project/mvp2/menu/menu_data/menu_item_model.dart';
+
+/// Expose menu item related operations
+final menuService = MenuService();
+
+class MenuService {
+  final SupabaseClient _supabase;
+  final bool isTest;
+  final String table;
+
+  MenuService({SupabaseClient? client, this.isTest = false})
+    : _supabase = client ?? Supabase.instance.client,
+      table = _resolveTable(isTest);
+
+  static String _resolveTable(bool isTest) {
+    if (isTest) {
+      return dotenv.env["MENU_TABLE_STAGING"] ?? '';
+    }
+    const prodTable = String.fromEnvironment('MENU_TABLE_PROD');
+    return prodTable.isNotEmpty
+        ? prodTable
+        : dotenv.env["MENU_TABLE_PROD"] ?? '';
+  }
+
+  Future<void> insertMenuItem(MenuItem item) async {
+    try {
+      Map<String, dynamic> payload = item.toMap();
+      debugPrint('Inserting menu item: $payload');
+      final response = await _supabase.from(table).insert(payload);
+      print("Insert response: $response");
+    } catch (e) {
+      print('Insert menu item error: $e');
+      throw Exception('Failed to insert menu item');
+    }
+  }
+
+  Future<List<MenuItem>> getUserMenuItems(String userId) async {
+    try {
+      final res = await _supabase.from(table).select().eq('user_id', userId);
+
+      // res is List<dynamic>
+      return res
+          .map((row) => MenuItem.fromMap(row as Map<String, dynamic>))
```

Comment:
Casting is unnecessary, `fromMap()` already returns type `Map<String, dynamic>`.

```suggestion
          .map((row) => MenuItem.fromMap(row))
```